actionpack/lib/action_controller/metal/http_authentication.rb

```diff
@@ -94,13 +94,17 @@ def authenticate(request, &login_procedure)
       end
 
       def has_basic_credentials?(request)
-        request.authorization.present? && (auth_scheme(request).downcase == 'basic')
+        request.authorization.present? && (auth_scheme(request).downcase == 'basic') && !user_name_or_password_blank?(request)
       end
 
       def user_name_and_password(request)
         decode_credentials(request).split(':', 2)
       end
 
+      def user_name_or_password_blank?(request)
```

Comment:
Good point. To be honest, I just rebased Lau's branch and did really looked at the implementation.
In my opinion we should not continue the login process if either username or password is blank.
Any opinion?